src/mappings/core.ts

```diff
@@ -586,3 +594,24 @@ export function handleSwap(event: Swap): void {
   // update unique day swaps
   addDailyUniqueAddressInteraction(event, swap.from)
 }
+
+export function handleBlockPairTokenPrice(block: ethereum.Block): void {
+  let pairContractAddress = dataSource.address()
+  let pair = Pair.load(pairContractAddress.toHexString())
+
+  if (!pair) {
+    return
+  }
+
+  let blockPairTokenPrice = new BlockPairTokenPrice(block.hash.toHexString())
```

Comment:
For each block, the total number of `BlockPairTokenPrice` entities created equals the number of Swapr pairs the protocol has at that block.

In this implementation, all BlockPairTokenPrice share the same ID which is an issue. A simple uniqueness solution is using block hash or block height with the pair address:


```suggestion
  let blockPairTokenPrice = new BlockPairTokenPrice(block.number.toString() + '-' + pairContractAddress.toHexString())
```

This produces IDs like: `15061515-0x1865d5445010e0baf8be2eb410d3eae4a68683c2`, `15061516-0x1865d5445010e0baf8be2eb410d3eae4a68683c2`, `15061517-0x1865d5445010e0baf8be2eb410d3eae4a68683c2` and so on

Reply:
Thanks Adam, it's now fixed in [fe8aeca](https://github.com/dxgraphs/swapr-subgraph/pull/23/commits/fe8aecae915fe290462a0f652681a4a6f51b3575)